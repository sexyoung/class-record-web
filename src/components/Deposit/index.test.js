import { render, screen } from '@testing-library/react';
import UserEvent from "@testing-library/user-event";
import { Deposit } from '.';

const postDeposit = (({ planId, studentId, date }) =>
  Promise.resolve({})
);

const planList = [{
  createdAt: "2021-10-28 22:29:12",
  expiresDays: 56,
  id: 2,
  isActive: true,
  money: 2000,
  name: "五堂套票",
  teacherId: 1,
  times: 5,
  updatedAt: "2021-10-28 22:29:12"
},
{
  createdAt: "2021-10-28 22:29:12",
  expiresDays: 84,
  id: 3,
  isActive: true,
  money: 2400,
  name: "八堂套票",
  teacherId: 1,
  times: 8,
  updatedAt: "2021-10-28 22:29:12"
}];

const student = {
  email: "hfjdjd@djdur.djd",
  expiresAt: "2022-02-13 15:04:51",
  id: 28,
  isExpired: false,
  lineId: null,
  name: "WALLET",
  phone: "0930123456",
  picture: "e52dfdb11ca1691e0e266f4e82a428a9.png",
  planTimes: 8,
  rollcalls: 1,
  status: "join",
  times: 7
};

jest.mock('api', () =>  {
  return { postDeposit };
});

it('should onSubmit be triggered', () => {
  const { getByTestId, getByRole } = render(<Deposit
    planList={planList}
    student={student}
    closeModal={() => {}}
    depositDone={() => {}}
  />);

  UserEvent.type(getByTestId('2'), '2');
  UserEvent.type(getByTestId('date'), '2022-01-02');
  UserEvent.click(getByRole('button'));

  expect(getByTestId('2').value).toBe('2');
  expect(getByTestId('date').value).toBe('2022-01-02');
});

