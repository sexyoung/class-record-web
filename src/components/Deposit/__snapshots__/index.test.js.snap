// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should onSubmit be triggered 1`] = `
<div>
  <form
    class="Deposit"
  >
    <div
      class="title"
    >
      儲值
    </div>
    <div
      class="img"
    />
    <div
      class="name"
    >
      WALLET
    </div>
    <select
      class="input mt-5"
      name="plan"
    >
      <option
        data-testid="2"
        value="2"
      >
        五堂套票
      </option>
      <option
        data-testid="3"
        value="3"
      >
        八堂套票
      </option>
    </select>
    <div
      class="inputTitle"
    >
      儲值日期
    </div>
    <input
      class="input mt-0"
      data-testid="date"
      name="date"
      required=""
      type="date"
    />
    <button
      class="btn w-full mt-3"
    >
      儲值
    </button>
  </form>
</div>
`;
